sunpy/net/helioviewer.py

```diff
@@ -450,3 +435,15 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, key):
+        """
+        Returns source_id based on the key.
+        """
+        source_id_list = list(partial_match(key,self.data_sources))
+        if len(source_id_list) == 1:
+            return source_id_list[0]
+        else:
+            raise KeyError(f"The values used: {key} do not correspond to one source_id "
+                            "but "+ str(len(source_id_list)) + " source_id(s)."
+                            " Please check the list using HelioviewerClient.data_sources.")
```

Comment:
I would rewrite this in a different way...
```python
if len(source_id_list) > 1: # or maybe != 1
    raise KeyError...
return source_id_list[0]
```
I prefer this way because it's obvious that the error message is for the case when there's more than one possibility (or not 1). As it is now it could make one think that you can change this by a `try/except` block.